skimage/feature/tests/test_util.py

```diff
@@ -86,3 +87,74 @@ def test_plot_matches():
         plot_matches(
             ax, img1, img2, keypoints1, keypoints2, matches, alignment='vertical'
         )
+
+
+@pytest.mark.skipif(not has_mpl, reason="Matplotlib not installed")
+def test_plot_matched_features():
+    from matplotlib import pyplot as plt
+
+    fig, ax = plt.subplots()
+
+    shapes = (
+        ((10, 10), (10, 10)),
+        ((10, 10), (12, 10)),
+        ((10, 10), (10, 12)),
+        ((10, 10), (12, 12)),
+        ((12, 10), (10, 10)),
+        ((10, 12), (10, 10)),
+        ((12, 12), (10, 10)),
+    )
+
+    keypoints0 = 10 * np.random.rand(10, 2)
+    keypoints1 = 10 * np.random.rand(10, 2)
+    idxs0 = np.random.randint(10, size=10)
+    idxs1 = np.random.randint(10, size=10)
+    matches = np.column_stack((idxs0, idxs1))
+
+    for shape0, shape1 in shapes:
+        img0 = np.zeros(shape0)
+        img1 = np.zeros(shape1)
+        plot_matched_features(
+            img0,
+            img1,
+            keypoints0=keypoints0,
+            keypoints1=keypoints1,
+            matches=matches,
+            ax=ax,
+        )
+        plot_matched_features(
+            img0,
+            img1,
+            ax=ax,
+            keypoints0=keypoints0,
+            keypoints1=keypoints1,
+            matches=matches,
+            only_matches=True,
+        )
+        plot_matched_features(
+            img0,
+            img1,
+            ax=ax,
+            keypoints0=keypoints0,
+            keypoints1=keypoints1,
+            matches=matches,
+            keypoints_color='r',
+        )
+        plot_matched_features(
+            img0,
+            img1,
+            ax=ax,
+            keypoints0=keypoints0,
+            keypoints1=keypoints1,
+            matches=matches,
+            matches_color='r',
+        )
```

Comment:
I think there's some repetition here? In my book it would be okay to shorten this even further and use `pytest.mark.parametrize` for the optional parameters.

Reply:
> I think there's some repetition here?

The difference is that one call tests for `keypoints_color='r'` and the other `matches_color='r'`.

> In my book it would be okay to shorten this even further and use `pytest.mark.parametrize` for the optional parameters.

Hmm, that would only increase the number of tests, since `pytest.mark.parametrize` lets you test different values for a given parameter. Here, only one value (per optional parameter) is being tested...

Reply:
Huh, I should check my eyes (or maybe take a nap). :sweat_smile: Any reason not to increase the number of tests? They don't seem slow on my machine.

Reply:
Oh, I'm not opposed to it, I just didn't think it was necessary and I thought (I guess I misunderstood) you were saying there were too many already!